src/voice-broadcast/stores/VoiceBroadcastRecordingStore.ts

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { MatrixClient, MatrixEvent, RelationType } from "matrix-js-sdk/src/matrix";
+import { TypedEventEmitter } from "matrix-js-sdk/src/models/typed-event-emitter";
+
+import { VoiceBroadcastInfoEventType, VoiceBroadcastInfoState } from "..";
+
+export enum VoiceBroadcastRecordingStoreEvent {
+    StateChanged = "liveness_changed",
+}
+
+interface EventMap {
+    [VoiceBroadcastRecordingStoreEvent.StateChanged]: (state: VoiceBroadcastInfoState) => void;
+}
+
+/**
+ * This store handles one Voice Broadcast recording.
+ */
+export class VoiceBroadcastRecordingStore extends TypedEventEmitter<VoiceBroadcastRecordingStoreEvent, EventMap> {
+    private _state: VoiceBroadcastInfoState;
+
+    public constructor(
+        public readonly infoEvent: MatrixEvent,
+        private client: MatrixClient,
+    ) {
+        super();
+
+        const room = this.client.getRoom(this.infoEvent.getRoomId());
+        const relations = room?.getUnfilteredTimelineSet()?.relations?.getChildEventsForEvent(
+            this.infoEvent.getId(),
+            RelationType.Reference,
+            VoiceBroadcastInfoEventType,
+        );
+        const relatedEvents = relations?.getRelations();
+        this._state = !relatedEvents?.find((event: MatrixEvent) => {
+            return event.getContent()?.state === VoiceBroadcastInfoState.Stopped;
+        }) ? VoiceBroadcastInfoState.Started : VoiceBroadcastInfoState.Stopped;
+
+        // XXX add listening for updates
```

Comment:
You probably already have this in progress, but maybe add your name next to these TODOs